### PR TITLE
added multi-dof support for simple_controller_manager

### PR DESCRIPTION
**Pull request by hvpandya**
- Added multi_dof_follow_joint_trajectory_handle which exposes control_msgs/MultiDOFFollowJointTrajectoryAction to support multi-dof joints in moveit_simple_controller_manager.

### CODE REVIEW

File: moveit_simple_controller_manager/package.xml [hunk 1]
```diff
@@ -11,14 +11,14 @@
...
-  <buildtool_depend>catkin</buildtool_depend> 
+  <buildtool_depend>catkin</buildtool_depend>
 
   <build_depend>moveit_core</build_depend>
   <build_depend>roscpp</build_depend>
   <build_depend>pluginlib</build_depend>
   <build_depend>control_msgs</build_depend>
   <build_depend>actionlib</build_depend>
-
+
```
Comment (mikeferguson) [inline moveit_simple_controller_manager/package.xml]: this should be reverted

File: moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp [hunk 2]
```diff
@@ -145,6 +147,15 @@ class MoveItSimpleControllerManager : public moveit_controller_manager::MoveItCo
             controllers_[name] = new_handle;
           }
         }
+        else if(type == "MultiDOFFollowJointTrajectory")
```
Comment (mikeferguson) [inline moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp]: space after if

File: moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp [hunk 3]
```diff
@@ -145,6 +147,15 @@ class MoveItSimpleControllerManager : public moveit_controller_manager::MoveItCo
...
+        {
+          new_handle.reset(new MultiDOFFollowJointTrajectoryControllerHandle(name, action_ns));
+          if (static_cast<MultiDOFFollowJointTrajectoryControllerHandle*>(new_handle.get())->isConnected())
+	  {
```
Comment (mikeferguson) [inline moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp]: identation on this block is off

File: moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp [hunk 4]
```diff
@@ -155,10 +166,13 @@ class MoveItSimpleControllerManager : public moveit_controller_manager::MoveItCo
           controllers_.erase(name);
           continue;
         }
-        
+
         /* add list of joints, used by controller manager and moveit */
         for (int j = 0 ; j < controller_list[i]["joints"].size() ; ++j)
+        {
           controllers_[name]->addJoint(std::string(controller_list[i]["joints"][j]));
+          ROS_INFO_STREAM("\n"<<std::string(controller_list[i]["joints"][j])<<"\n");
```
Comment (mikeferguson) [inline moveit_simple_controller_manager/src/moveit_simple_controller_manager.cpp]: this is probably pretty spammy and should be removed or changed to a DEBUG_NAMED

File: moveit_simple_controller_manager/include/moveit_simple_controller_manager/multi_dof_follow_joint_trajectory_handle.h [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+public:
+
+  MultiDOFFollowJointTrajectoryControllerHandle(const std::string &name, const std::string &action_ns) :
+    ActionBasedControllerHandle<control_msgs::MultiDOFFollowJointTrajectoryAction>(name, action_ns)
+  {
+  }
+
+  virtual bool sendTrajectory(const moveit_msgs::RobotTrajectory &trajectory)
+  {
+    ROS_DEBUG_STREAM("MultiDOFFollowJointTrajectoryController: new trajectory to " << name_);
```
Comment (mikeferguson) [inline moveit_simple_controller_manager/include/moveit_simple_controller_manager/multi_dof_follow_joint_trajectory_handle.h]: debugs should generally be debug_named so that they can be enabled more easily

File: moveit_simple_controller_manager/include/moveit_simple_controller_manager/follow_joint_trajectory_controller_handle.h [hunk 6]
```diff
@@ -64,9 +63,14 @@ class FollowJointTrajectoryControllerHandle : public ActionBasedControllerHandle
     if (!controller_action_client_)
       return false;
 
+    /*
+     * This should never happen as the moveit_simple_controller_manager now loads
+     * MULTI_DOF_FOLLOW_JOINT_CONTROLLER_HANDLE for multi-dof joints
+     */
```
Comment (mikeferguson) [inline moveit_simple_controller_manager/include/moveit_simple_controller_manager/follow_joint_trajectory_controller_handle.h:69]: this is a misleading comment -- this should not happen unless someone misconfigures their controllers -- which is exactly what this check is here for.

File: moveit_simple_controller_manager/include/moveit_simple_controller_manager/follow_joint_trajectory_controller_handle.h [hunk 7]
```diff
@@ -64,9 +63,14 @@ class FollowJointTrajectoryControllerHandle : public ActionBasedControllerHandle
...
     if (!trajectory.multi_dof_joint_trajectory.points.empty())
     {
-      ROS_WARN("FollowJointTrajectoryController: %s cannot execute multi-dof trajectories.", name_.c_str());
+      ROS_ERROR("FollowJointTrajectoryController: %s cannot execute multi-dof trajectories. Check moveit_simple_controller_manager to load MOVEIT_PLUGINS_MULTI_DOF_FOLLOW_JOINT_CONTROLLER_HANDLE correctly", name_.c_str());
+      return false;
```
Comment (mikeferguson) [inline moveit_simple_controller_manager/include/moveit_simple_controller_manager/follow_joint_trajectory_controller_handle.h]: please revert these changes -- we can't apply them to indigo at this point (could change the behavior of existing configurations)

File: moveit_simple_controller_manager/include/moveit_simple_controller_manager/multi_dof_follow_joint_trajectory_handle.h [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+    if (!controller_action_client_)
+      return false;
+
+      /*
+       * This should never happen as the moveit_simple_controller_manager loads
+       * MOVEIT_PLUGINS_FOLLOW_TRAJECTORY_CONTROLLER_HANDLE for single-dof joints
+       */
+    if (!trajectory.joint_trajectory.points.empty())
```
Comment (mikeferguson) [inline moveit_simple_controller_manager/include/moveit_simple_controller_manager/multi_dof_follow_joint_trajectory_handle.h:72]: again, this comment is misleading -- if the controllers.yaml is wrong, this will hit.

**Comment by hvpandya** [comment]
Fixed the said issues and applied the suggestions.

**Comment by mikeferguson** [comment]
Depends on https://github.com/ros-controls/control_msgs/pull/16
